### PR TITLE
[Linux] Initial implementation of fabric-bridge for fabric sync

### LINKED ISSUE #33264: [Feature] Create Fabric-Bridge Example App that implements the Aggregator device type
### Feature description

...
Develop an example application that implements the Aggregator device type. This app will act as a bridge device and facilitate Fabric Synchronization.

### Platform

all, other

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yufengwangca**
An example application to implement the Aggregator device type for Fabric Synchronization and demonstrates the end-to-end Fabric Synchronization feature using dynamic endpoints.

Fixes #33264

### CODE REVIEW

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter [hunk 1]
```diff
@@ -0,0 +1,2035 @@
...
+    callback attribute specificationVersion;
+    callback attribute maxPathsPerInvoke;
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 3;
+  }
+
+  server cluster LocalizationConfiguration {
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1703]: Does this app actually implement this cluster usefully?

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter [hunk 2]
```diff
@@ -0,0 +1,2035 @@
...
+    persist  attribute activeLocale default = "en-US";
+    callback attribute supportedLocales;
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
+
+  server cluster TimeFormatLocalization {
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1713]: Does this app actually implement this cluster usefully?

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter [hunk 3]
```diff
@@ -0,0 +1,2035 @@
...
+    persist  attribute activeCalendarType default = 0;
+    callback attribute supportedCalendarTypes;
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
+
+  server cluster UnitLocalization {
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1724]: Does this app actually implement this cluster usefully?

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter [hunk 4]
```diff
@@ -0,0 +1,2035 @@
...
+
+    handle command KeySetWrite;
+    handle command KeySetRead;
+    handle command KeySetReadResponse;
+    handle command KeySetRemove;
+    handle command KeySetReadAllIndices;
+    handle command KeySetReadAllIndicesResponse;
+  }
+
+  server cluster UserLabel {
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1984]: Does this app use this cluster?

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter [hunk 5]
```diff
@@ -0,0 +1,2035 @@
...
+    callback attribute labelList;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
+}
+endpoint 1 {
+  device type ma_aggregator = 14, version 1;
+
+  binding cluster Binding;
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1993]: Why does this have a Binding client?

File: examples/fabric-bridge-app/linux/ActionsServerCallbacks.cpp [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        }
+        return CHIP_NO_ERROR;
+    });
+    return err;
+}
+
+CHIP_ERROR ActionsAttrAccess::ReadSetupUrlAttribute(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    static const char SetupUrl[] = "https://example.com";
+    return aEncoder.Encode(chip::CharSpan::fromCharString(SetupUrl));
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/linux/ActionsServerCallbacks.cpp:104]:
```suggestion
    return aEncoder.Encode("https://example.com"_span);
```

File: examples/fabric-bridge-app/linux/ActionsServerCallbacks.cpp [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+#include "main.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Actions::Attributes;
+
+namespace {
+
+class ActionsAttrAccess : public AttributeAccessInterface
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/linux/ActionsServerCallbacks.cpp:40]: Do you actually want Actions on the fabric bridge?  What actions would you support, exactly?

File: examples/fabric-bridge-app/linux/main.cpp [hunk 8]
```diff
@@ -0,0 +1,381 @@
...
+
+namespace {
+
+const int kNodeLabelSize = 32;
+// Current ZCL implementation of Struct uses a max-size array of 254 bytes
+const int kDescriptorAttributeArraySize = 254;
+
+EndpointId gCurrentEndpointId;
+EndpointId gFirstDynamicEndpointId;
+// Power source is on the same endpoint as the composed device
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/linux/main.cpp:71]: This comment does not make sense.  There is no power source anywhere here.

File: examples/fabric-bridge-app/linux/main.cpp [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+Device * gDevices[CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT + 1];
+std::vector<Room *> gRooms;
+std::vector<Action *> gActions;
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/linux/main.cpp:74]: So these are always empty, yes?  Why do we have them at all?

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter (hunk 1 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1703]: Disable until we actually use

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter (hunk 2 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1713]: Disable until we actually use

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter (hunk 3 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1724]: Disable until we actually use

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter (hunk 5 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter:1993]: Disable until we actually use

File: examples/fabric-bridge-app/linux/ActionsServerCallbacks.cpp (hunk 7 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/linux/ActionsServerCallbacks.cpp]: A Bridge typically has information on topology or logical grouping of the Bridged Devices. If a Bridge has such information on topology or logical grouping, it SHOULD expose such information in the EndpointLists attribute of an ActionsClusterluster.

We can ignore the ActionsCluster for now, until we use it.

File: examples/fabric-bridge-app/linux/main.cpp (hunk 9 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/linux/main.cpp]: This is to support bridge topology, we can ignore it for now, until we use it.

File: examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter (hunk 4 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/fabric-bridge-common/fabric-bridge-app.matter]: Disable until we actually use

File: examples/fabric-bridge-app/linux/main.cpp [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+    uint8_t index = 0;
+    while (index < CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT)
+    {
+        if (nullptr == gDevices[index])
+        {
+            gDevices[index] = dev;
+            CHIP_ERROR err;
+            while (true)
+            {
+                // Todo: Update this to schedule the work rather than use this lock
```
Comment (tehampson) [inline examples/fabric-bridge-app/linux/main.cpp:94]: Is there a bug associated with this TODO? Is this going to be addressed in a future PR?

**Review by tehampson — APPROVED** [review]

File: examples/fabric-bridge-app/linux/main.cpp (hunk 10 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/linux/main.cpp]: This is to address potential infinite loop if no valid endpoint ID can be found and no errors other than CHIP_ERROR_ENDPOINT_EXISTS are encountered.

I adding a maximum number of retries to exit the loop more reliably instead.

File: examples/fabric-bridge-app/linux/main.cpp [hunk 11]
```diff
@@ -0,0 +1,322 @@
...
+using namespace chip::app;
+using namespace chip::Credentials;
+using namespace chip::Inet;
+using namespace chip::Transport;
+using namespace chip::DeviceLayer;
+using namespace chip::app::Clusters;
+
+namespace {
+
+const int kNodeLabelSize = 32;
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/linux/main.cpp:63]: kNodeLabelSize looks unused.

File: examples/fabric-bridge-app/linux/main.cpp [hunk 12]
```diff
@@ -0,0 +1,322 @@
...
+// Current ZCL implementation of Struct uses a max-size array of 254 bytes
+const int kDescriptorAttributeArraySize = 254;
```
Comment (bzbarsky-apple) [inline examples/fabric-bridge-app/linux/main.cpp:65]: This comment seems unrelated to anything, and kDescriptorAttributeArraySize is unused, yes?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/fabric-bridge-app/linux/main.cpp (hunk 11 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/linux/main.cpp]: Acked

File: examples/fabric-bridge-app/linux/main.cpp (hunk 12 above)
Comment (yufengwangca) [reply examples/fabric-bridge-app/linux/main.cpp]: Acked
